### PR TITLE
Replace `DateTime` with `DateTimeImmutable`

### LINKED ISSUE #21: DataTime imutable
One more suggestion - replace `DateTime` with [`DateTimeImmutable`](https://www.php.net/manual/en/class.datetimeimmutable.php) in all code places. Reason - date/time seems like immutable during API calls

### PR DESCRIPTION
**Pull request by gomzyakov**
### Changed

- Replace `DateTime` with `DateTimeImmutable` in all code places

Fixes #21 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

File: tests/Unit/Responses/Entities/EntitiesFactory.php
Comment (gomzyakov) [inline tests/Unit/Responses/Entities/EntitiesFactory.php:98]: I can get refactoring of this code & similar code below.

But, this is beyond the scope of the task, right?

File: tests/Unit/ClientTest.php
Comment (eldario) [inline tests/Unit/ClientTest.php]: del `()`

File: tests/Unit/ClientTest.php
Comment (eldario) [inline tests/Unit/ClientTest.php]: del `()`

**Review by eldario — CHANGES_REQUESTED** [review]

File: tests/Unit/ClientTest.php
Comment (gomzyakov) [reply tests/Unit/ClientTest.php]: Fixed 6b3bd06c0b6414c6ee11bc614c2003c1b22a17ee

File: tests/Unit/ClientTest.php
Comment (gomzyakov) [reply tests/Unit/ClientTest.php]: Fixed 6b3bd06c0b6414c6ee11bc614c2003c1b22a17ee

**Comment by tarampampam** [comment]
Blocked until #22 is not merged

**Pull request closed without merging**
